File: server/src/models/recipeModel.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const recipeSchema = new Schema({
  Contient: {
    type: String,
  },
  Country_State: {
    type: String,
  },
  cuisine: {
    type: String,
  },
  title: {
    type: String,
  },
  URL: {
    type: String,
  },
  rating: {
    type: Number,
  },
  total_time: {
    type: Number,
  },
  prep_time: {
    type: Number,
  },
  cook_time: {
    type: Number,
  },
  description: {
    type: String,
  },
  ingredients: [
    {
      type: String,
    },
  ],
  instructions: [
    {
      type: String,
    },
  ],
  nutrients: {
    calories: { type: String },
    carbohydrateContent: { type: String },
    cholesterolContent: { type: String },
    fiberContent: { type: String },
    proteinContent: { type: String },
    saturatedFatContent: { type: String },
    sodiumContent: { type: String },
    sugarContent: { type: String },
    fatContent: { type: String },
    unsaturatedFatContent: { type: String },
  },
  serves: {
    type: String,
  },
});

const Recipe = mongoose.model("Recipe", recipeSchema);

module.exports = Recipe;

File: backend/models/recipeModel.py
from mongoengine import Document, StringField, IntField, ListField, EmbeddedDocument, EmbeddedDocumentField

class Nutrients(EmbeddedDocument):
    calories = StringField()
    carbohydrateContent = StringField()
    cholesterolContent = StringField()
    fiberContent = StringField()
    proteinContent = StringField()
    saturatedFatContent = StringField()
    sodiumContent = StringField()
    sugarContent = StringField()
    fatContent = StringField()
    unsaturatedFatContent = StringField()

class Recipe(Document):
    Contient = StringField()
    Country_State = StringField()
    cuisine = StringField()
    title = StringField()
    URL = StringField()
    rating = IntField()
    total_time = IntField()
    prep_time = IntField()
    cook_time = IntField()
    description = StringField()
    ingredients = ListField(StringField())
    instructions = ListField(StringField())
    nutrients = EmbeddedDocumentField(Nutrients)
    serves = StringField()
